### PR TITLE
Smc stub set and interesting transitions

### PR DESCRIPTION
**Pull request by faje18**

### CODE REVIEW

**Review by rherum18 — APPROVED** [review]

File: include/PetriEngine/Stubborn/StubbornSet.h [hunk 1]
```diff
@@ -101,6 +101,8 @@ namespace PetriEngine {
             _queries = conds;
         }
 
+        virtual void addToStub(uint32_t t);
+
```
Comment (faje18) [inline include/PetriEngine/Stubborn/StubbornSet.h:105]: why was this moved? anybody knows?

Comment (MikkelTJensen) [reply include/PetriEngine/Stubborn/StubbornSet.h:105]: Because it was protected. LOL!

**Review by fulei345 — APPROVED** [review]
